### PR TITLE
Extract mod midi import export

### PR DESCRIPTION
**Pull request by Paul-Licameli**
Resolves: #5623

A module for import and export of MIDI, completing the set of import-export modules.

It will depend on Audacity, as some of the other existing modules do.

This will break some dependencies that are blocking the extraction of ProjectManager and ProjectFileManager into toolkit neutral libraries.

This unifies import code so that it's agnostic between audio and MIDI (see first QA item), but not Export of MIDI, which still requires a separate menu item.

QA:
- [x] Known behavior change! : File > Import > Audio will also import MIDI tracks.  (So should we keep the separate "Import MIDI" menu item?)
- [x] Drag-and-drop of MIDI files onto the project window
- [x] Import LOF applied to a MIDI file
- [x] MIDI files still appear in the File > Open Recent submenu
- [x] Imported MIDI tracks are correctly muted when the project had a Solo track (imported with either menu item)
- [x] Preferences > Import/Export Preference page behaves as before (all of the radio items) and sequences the pages as before
- [x] Disable mod-midi-import-export in Modules preferences; restart; observe Import/Export preferences again; reenable; restart; observe preferences once more; check sequence of pages and of the controls in the Import/Export page.

- [x] I signed [CLA](https://www.audacityteam.org/cla/)
- [x] The title of the pull request describes an issue it addresses
- [x] If changes are extensive, then there is a sequence of easily reviewable commits
- [x] Each commit's message describes its purpose and effects
- [x] There are no behavior changes unnecessary for the stated purpose of the PR

Recommended:
- [x] Each commit compiles and runs on my machine without known undesirable changes of behavior

### CODE REVIEW

**Comment by crsib** [comment]
This PR doesn't build on Windows. I think some classes are missing `_API` macros.

File: libraries/lib-import-export/Import.cpp [hunk 1]
```diff
@@ -173,9 +172,9 @@ bool Importer::Initialize()
    using namespace Registry;
    static OrderingPreferenceInitializer init{
       PathStart,
-      { {wxT(""), wxT("AUP,PCM,OGG,Opus,FLAC,MP3,LOF,WavPack,FFmpeg") } }
-      // QT and GStreamer are only conditionally compiled and would get
-      // placed at the end if present
+      // FFmpeg is in default of all other modules that might handle a format
+      // better and specially; including MIDI import
+      { {wxT(""), wxT("AUP,PCM,OGG,FLAC,MP3,LOF,WavPack,portsmf,portsmf,FFmpeg") } }
```
Comment (crsib) [inline libraries/lib-import-export/Import.cpp:177]: portsmf is listed twice

**Review by crsib — APPROVED** [review]
It's good to see MIDI import/export as a module. Please fix link issues first

**Comment by Paul-Licameli** [comment]
Rebased to recent master, easy conflicts resolved, duplicate "portsmf" fixed

**Comment by crsib** [comment]
Linking on Windows still fails.

**Comment by Paul-Licameli** [comment]
A one-work commit added to fix Windows link

**Review by crsib — APPROVED** [review]

**Comment by chinakov** [comment]
Tested, thank you!

**Pull request merged**